igor-web/src/main/java/com/netflix/spinnaker/igor/concourse/client/ConcourseClient.java

```diff
@@ -110,12 +146,20 @@ private void refreshTokenIfNecessary() {
 
   private Token refreshToken() {
     token =
-        tokenService.passwordToken(
-            "password", user, password, "openid profile email federated:id groups");
+        tokenServiceV1 != null
+            ? tokenServiceV1.passwordToken(
+                "password", user, password, "openid profile email federated:id groups")
+            : tokenServiceV2.passwordToken(
+                "password", user, password, "openid profile email federated:id groups");
+
     tokenExpiration = token.getExpiry();
     return token;
   }
 
+  public Response userInfo() {
+    return skyServiceV1 != null ? skyServiceV1.userInfo() : skyServiceV2.userInfo();
+  }
```

Comment:
This feels a bit dirty to me. I'd consider creating a new `SkyService` interface that both V1 and V2 extends from, given that they both just contains the `userInfo()` method signature. Then you can have just a single `skyService` field, and not care about the implementation and possible `null`-values here.
And then do exactly the same with `TokenService`.

Reply:
Example:
```
public interface SkyServiceV2 extends SkyService {
  @GET("/api/v1/user")
  @Override
  Response userInfo();
}
```

Reply:
fully agree! unfortunately, the library in use here, resteasy, doesn't allow extending interfaces :(

Reply:
see https://github.com/square/retrofit/issues/504 (appears this may have been fixed in a recent version; the one in use here is quite old)

Reply:
Oh, I wasn't aware... Yuck. 😛
Maybe put a little comment in the code with that explanation?